walrus_mainboard_driver/src/walrus_mainboard_driver/walrus_mainboard_driver.cpp

```diff
@@ -0,0 +1,670 @@
+	}
+	
+	bool MainBoardDriver::init()
+	{	    
+	    //Setup publishers and subscribers to communicate with the embedded board
+		hs_control = nh.advertise<walrus_firmware_msgs::MainBoardHighSpeedControl>("/walrus/main_board/hs_control", 1000);
+		to_board = nh.advertise<walrus_firmware_msgs::MainBoardControl>("/walrus/main_board/PC_to_board_control", 1000);
+		hs_feedback = nh.subscribe("/walrus/main_board/hs_feedback", 1000, &MainBoardDriver::hs_feedback_callback, this);
+		ls_data = nh.subscribe("/walrus/main_board/ls_data", 1000, &MainBoardDriver::ls_data_callback, this);
+		from_board = nh.subscribe("/walrus/main_board/board_to_PC_control", 1000, &MainBoardDriver::from_board_callback, this);
+	    return true;
+	}
+	
+	void MainBoardDriver::read()
+	{
+        double temp, dr;
+	    double interval = (ros::Time::now() - last_read).toSec();
+	    last_read = ros::Time::now();
+	    
+	    {
+	        boost::lock_guard<boost::mutex> lock(control_data_mutex);
+	       
+	        for (int l = 0; l < 4; l++)
+	        {
+		        temp = (hs_feedback_msg.pod_position[l]/1024.0)*2*PI;
+		        if (POD_REV[l])
+		            temp = (2*PI) - temp;
+		        temp -= POD_POSITION_NEUTRAL[l];
+		        if (temp < 0)
+		            temp += 2*PI;
+		        dr = temp - pod_position[l];
+		        if (dr > 0 && dr > PI) //Decrease angle across 0
+		            pod_velocity[l] = (dr-2*PI)/interval;
+		        else if (dr < 0 && dr < -PI) //Increase angle across 0
+		            pod_velocity[l] = (dr+2*PI)/interval;
+		        else  
+		            pod_velocity[l] = dr/interval;
+		        pod_position[l] = temp;		
+		        pod_effort[l] = (hs_feedback_msg.motor_current[l] / 1000.0) * 29.184; //RS550 motors with 5120:1 reduction outputs 29.185 Nm/A
+	        }		
+	    }
+
+	}
+	
+	void MainBoardDriver::write()
+	{			
+	    walrus_firmware_msgs::MainBoardHighSpeedControl hs_control_msg;    
+	    {
+	        boost::lock_guard<boost::mutex> lock(control_data_mutex);
+	        //Send motor control message	        
+	        for (int l = 0; l < 4; l++)
+	        {
+	            pod_power[l] = (pod_effort_cmd[l]*250) + 1500;
+		        hs_control_msg.motor_power[l] = pod_power[l];
+		    }
+		}
+		
+		hs_control.publish(hs_control_msg);		
+	}
+	
+	void MainBoardDriver::update_diagnostics()
+	{
+		diagnostic_updater.update();
+	}
+	
+	//Subscription callbacks
+	void MainBoardDriver::hs_feedback_callback(const walrus_firmware_msgs::MainBoardHighSpeedFeedback& msg)
+	{
+	    boost::lock_guard<boost::mutex> lock(control_data_mutex);
+	    hs_feedback_msg = msg;
+	    last_hs_msg = ros::Time::now();
+	}
+	void MainBoardDriver::ls_data_callback(const walrus_firmware_msgs::MainBoardLowSpeedData& msg)
+	{
+	    boost::lock_guard<boost::mutex> lock(sensor_data_mutex);
+        
+        //Temp sensors
+        for (int l = 0; l < 10; l++)
+            temp_sensors[l] = msg.temp_sense[l] / 10.0;
```

Comment:
Where do the 10, 100 and 1000
If these are different units can you document what the type is in the message and what the resultant value type is like here: https://github.com/RIVeR-Lab/epos_hardware/blob/indigo-devel/epos_hardware/src/util/epos.cpp#L203